website/static/js/nodesPrivacy.js

```diff
@@ -0,0 +1,272 @@
+var NodesPrivacyTreebeard = require('js/nodesPrivacySettingsTreebeard');
+
+//Todo (billyhunt): change url to dev or production as needed
+var API_BASE = 'http://localhost:8000/v2/nodes/';
+
+var MESSAGES = {
+    makeProjectPublicWarning:
+                        'Please review your projects, components, and add-ons for sensitive or restricted information before making them public.' +
+                        '<br><br>Once they are made public, you should assume they will always be public. You can ' +
+                        'return them to private later, but search engines (including Google’s cache) or others may access files before you do.',
+
+    selectNodes: 'Adjust your privacy settings by checking the boxes below. ' +
+                        '<br><br><b>Checked</b> projects and components will be <b>public</b>.  <br><b>Unchecked</b> components will be <b>private</b>.',
+    addonWarning: {
+        nodesPublic: 'The following projects and components will be made <b>public</b>.',
+        nodesPrivate: 'The following projects and components will be made <b>private</b>.'
+    }
+};
+
+function getNodesOriginal(nodeTree, nodesOriginal) {
+    /**
+     * take treebeard tree structure of nodes and get a dictionary of parent node and all its
+     * children
+     */
+    var i;
+    var nodeId = nodeTree.node.id;
+    nodesOriginal[nodeId] = {
+        public: nodeTree.node.is_public,
+        id: nodeTree.node.id,
+        addons: nodeTree.node.addons,
+        title: nodeTree.node.title,
+        canWrite: nodeTree.node.can_write,
+        changed: false
+    };
+
+    if (nodeTree.children) {
+        for (i in nodeTree.children) {
+            nodesOriginal = getNodesOriginal(nodeTree.children[i], nodesOriginal);
+        }
+    }
+    return nodesOriginal;
+}
+
+function patchNodesPrivacy(nodes) {
+    /**
+     * patches all the nodes in a changed state
+     * uses API v2 bulk requests
+     */
+    var nodesPatch = [];
+    var promise;
+    nodesPatch = nodes.splice(0,10).map(function (node) {
+        return {
+            'type': 'nodes',
+            'id': node.id,
+            'attributes': {
+                'public': node.public
+            }
+        };
+    });
+    return $.ajax({
+        url: API_BASE,
+        type: 'PATCH',
+        dataType: 'json',
+        contentType: 'application/vnd.api+json; ext=bulk',
+        crossOrigin: true,
+        xhrFields: {withCredentials: true},
+        processData: false,
+        data: JSON.stringify({
+            data: nodesPatch
+        })
+    }).done(function (response) {
+        if (nodes.length === 0) {
+            return;
+        }
+        return patchNodesPrivacy(nodes);
+    });
+}
+
+var NodesPrivacyViewModel = function(data, parentIsPublic) {
+    /**
+     * view model which corresponds to nodes_privacy.mako (#nodesPrivacy)
+     *
+     * @type {NodesPrivacyViewModel}
+     */
+    var self = this;
+    var nodesOriginal = {};
```

Comment:
Looking closer, the code no longer needs the self.nodesOriginal() knockout variable, so nodesOriginal is whats hot, and self.nodesOriginal() is dead to me, and hence deleted.